Tests/PHPUnit/Services/Mollie/Payments/Extractor/ApiExceptionDetailsExtractorTest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace MollieShopware\Tests\Services\Mollie\Payments\Extractor;
+
+use Mollie\Api\Exceptions\ApiException;
+use MollieShopware\Services\Mollie\Payments\Extractor\ApiExceptionDetailsExtractor;
+use MollieShopware\Services\Mollie\Payments\Models\PaymentFailedDetails;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @coversDefaultClass \MollieShopware\Services\Mollie\Payments\Extractor\ApiExceptionDetailsExtractor
+ */
+class ApiExceptionDetailsExtractorTest extends TestCase
+{
+    /**
+     * @var ApiExceptionDetailsExtractor
+     */
+    private $extractor;
+
+    protected function setUp(): void
+    {
+        $this->extractor = new ApiExceptionDetailsExtractor();
+    }
+
+    public function testFieldNotSet(): void
+    {
+        $exception = new ApiException();
+        $result = $this->extractor->extractDetails($exception);
+        $this->assertNull($result);
+    }
+
+    public function testMessageHasInvalidFormat(): void
+    {
+        $exception = new ApiException('test', 1, 'test.field');
+
+        $result = $this->extractor->extractDetails($exception);
+        $this->assertNull($result);
+    }
+
+    public function testMessageWithoutDocumentation(): void
+    {
+        $exception = new ApiException('Error executing API call (123:Test): Test Message', 123, 'test.field');
```

Comment:
could we have a separate test that explains us that the testfield will be extracted and added with an "Error" prefix in that object?
had a hard time figuring out

Reply:
i added here descriptions to the tests, i guess this is then more clear